From b6988652073900e3509b6cbe489cb6e4dec8a531 Mon Sep 17 00:00:00 2001
From: Lauren Post <lauren.post@freescale.com>
Date: Tue, 27 Aug 2013 12:15:59 -0500
Subject: [PATCH] yocto: Makefile changes to build on Yocto

Support for exclusion of libraries on mx6 rng and sahara

Signed-off-by: Lauren Post <lauren.post@freescale.com>
---
 hdmi-cec/Makefile    |    4 ++--
 ipu/Makefile         |    4 ++--
 pxp/Makefile         |    4 ++--
 rng/Makefile         |    9 ++++++++-
 sahara2/Makefile     |    9 ++++++++-
 screenlayer/Makefile |    4 ++--
 sim/Makefile         |    4 ++--
 7 files changed, 28 insertions(+), 14 deletions(-)

diff --git a/hdmi-cec/Makefile b/hdmi-cec/Makefile
index a1b7898..aebb0a8 100755
--- a/hdmi-cec/Makefile
+++ b/hdmi-cec/Makefile
@@ -1,5 +1,5 @@
-CC=$(CROSS_COMPILE)gcc
-AR=$(CROSS_COMPILE)ar
+CC ?=$(CROSS_COMPILE)gcc
+AR ?=$(CROSS_COMPILE)ar
 
 # list of platforms which want this test case
 INCLUDE_LIST:= IMX6Q
diff --git a/ipu/Makefile b/ipu/Makefile
index 2c22d72..deea1cf 100644
--- a/ipu/Makefile
+++ b/ipu/Makefile
@@ -1,5 +1,5 @@
-CC=$(CROSS_COMPILE)gcc
-AR=$(CROSS_COMPILE)ar
+CC ?=$(CROSS_COMPILE)gcc
+AR ?=$(CROSS_COMPILE)ar
 
 ifeq ($(PLATFORM), IMX6Q)
 OBJS = mxc_ipu_hl_lib_dummy.o
diff --git a/pxp/Makefile b/pxp/Makefile
index f2ca39b..d204c30 100644
--- a/pxp/Makefile
+++ b/pxp/Makefile
@@ -1,5 +1,5 @@
-CC=$(CROSS_COMPILE)gcc
-AR=$(CROSS_COMPILE)ar
+CC ?=$(CROSS_COMPILE)gcc
+AR ?=$(CROSS_COMPILE)ar
 
 # list of platforms which want this test case
 INCLUDE_LIST:=IMX50 IMX51 IMX5 IMX6Q IMX6S
diff --git a/rng/Makefile b/rng/Makefile
index 4b1dd5c..c0ccf02 100644
--- a/rng/Makefile
+++ b/rng/Makefile
@@ -1,6 +1,9 @@
+EXCLUDE_LIST:= IMX6Q
+
+ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 AR = ar -crv
 RM = rm -f
-CC=$(CROSS_COMPILE)gcc
+CC ?=$(CROSS_COMPILE)gcc
 LD=$(CROSS_COMPILE)ld
 
 INC := $(INCLUDE)
@@ -44,3 +47,8 @@ clean:
 fsl_shw_keystore.o: ../sahara2/fsl_shw_keystore.c
 	$(CC) $(CFLAGS) -c ../sahara2/fsl_shw_keystore.c
 
+else
+OBJS =
+all: ${OBJS}
+install: ${OBJS}
+endif
diff --git a/sahara2/Makefile b/sahara2/Makefile
index 106d932..20ff28f 100644
--- a/sahara2/Makefile
+++ b/sahara2/Makefile
@@ -1,6 +1,9 @@
+EXCLUDE_LIST:= IMX6Q
+
+ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 AR = ar -crv
 RM = rm -f
-CC=$(CROSS_COMPILE)gcc
+CC ?=$(CROSS_COMPILE)gcc
 LD=$(CROSS_COMPILE)ld
 
 INC := $(INCLUDE)
@@ -77,3 +80,8 @@ fsl_shw_keystore.o: fsl_shw_keystore.c
 um_adaptor.o: um_adaptor.c
 
 sf_util.o: sf_util.c
+else
+OBJS =
+all: ${OBJS}
+install: ${OBJS}
+endif

diff --git a/screenlayer/Makefile b/screenlayer/Makefile
index a90d4d5..bdebf24 100644
--- a/screenlayer/Makefile
+++ b/screenlayer/Makefile
@@ -1,5 +1,5 @@
-CC=$(CROSS_COMPILE)gcc
-AR=$(CROSS_COMPILE)ar
+CC ?=$(CROSS_COMPILE)gcc
+AR ?=$(CROSS_COMPILE)ar
 INC = $(INCLUDE) -I../ipu
 
 OBJS = ScreenLayer.o
diff --git a/sim/Makefile b/sim/Makefile
index 7685e32..45d43c0 100644
--- a/sim/Makefile
+++ b/sim/Makefile
@@ -1,5 +1,5 @@
-CC=$(CROSS_COMPILE)gcc
-AR=$(CROSS_COMPILE)ar
+CC ?=$(CROSS_COMPILE)gcc
+AR ?=$(CROSS_COMPILE)ar
 
 OBJS = iso7816-3.o
 
-- 
1.7.9.5

